### PR TITLE
#90 - Added enable/disable payment if no payment method selected

### PR DESCRIPTION
**Pull request by ahmed-shehata**
Issue #90  - Added enable/disable payment if no payment method selected

### CODE REVIEW

File: AtlasUI/AtlasUI/CheckoutSummaryStyler.swift [hunk 1]
```diff
@@ -136,15 +136,21 @@ class CheckoutSummaryStyler: LocalizerProviderType {
...
+        button.enabled = true
         button.translatesAutoresizingMaskIntoConstraints = false
         button.heightAnchor.constraintEqualToConstant(50).active = true
         button.topAnchor.constraintEqualToAnchor(viewController.termsAndConditionsButton.bottomAnchor,
             constant: -80).active = true
         button.leadingAnchor.constraintEqualToAnchor(viewController.view.leadingAnchor, constant: 10).active = true
         button.trailingAnchor.constraintEqualToAnchor(viewController.view.trailingAnchor, constant: -10).active = true
         let title = viewController.customer != nil ? "Buy Now" : "Connect To Zalando"
+
+        if (viewController.customer != nil) && (viewController.checkoutViewModel.checkout?.payment.selected == nil) {
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryStyler.swift]: Brackets are not needed here

File: AtlasUI/AtlasUI/CheckoutSummaryStyler.swift [hunk 2]
```diff
@@ -136,15 +136,21 @@ class CheckoutSummaryStyler: LocalizerProviderType {
...
+            button.backgroundColor = UIColor.grayColor()
+            button.enabled = false
+        }
         button.setTitle(title, forState: .Normal)
-        button.backgroundColor = UIColor.orangeColor()
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryStyler.swift]: Yes, could be done with ternary operator, but I'm getting older and older and prefer those old if-else.

```
private var isPaymentChosen: Bool {
  return viewController.customer != nil && viewController.checkoutViewModel.checkout?.payment.selected != nil
}

func updateMainButtonState() {
  if isPaymentChosen {
        button.backgroundColor = UIColor.orangeColor() 
        button.enabled = true
  } else {
        button.backgroundColor = UIColor. grayColor() 
        button.enabled = false
  }
```

Comment (ahmed-shehata) [inline AtlasUI/AtlasUI/CheckoutSummaryStyler.swift]: Much cleaner 👍 Thanks

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryStyler.swift [hunk 3]
```diff
@@ -13,8 +13,8 @@ class CheckoutSummaryStyler: LocalizerProviderType {
         viewController = checkoutSummaryViewController
     }
 
-    private var isPaymentChosen: Bool {
-        return viewController.customer != nil && viewController.checkoutViewModel.checkout?.payment.selected != nil
+    private var noPaymentSelected: Bool {
+        return viewController.customer != nil && viewController.checkoutViewModel.checkout?.payment.selected == nil
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryStyler.swift]: I suggested this `!=` on purpose. It seems broader condition to fulfil both: customer and selected payment, but don't know the details.

**Comment by bonkey** [comment]
👍

**Comment by ahmed-shehata** [comment]
👍

**Pull request merged**